### PR TITLE
fix: login and consent redirect behavior change since 1.9.x

### LINKED ISSUE #2363: redirect behavior change between v1.8.5 and v1.9.x
**Describe the bug**
The /oauth2/auth changed redirect behavior between v1.8.5 and v1.9.x. If the URLS_LOGIN is an angular UI app that contains /#/ e.g. http://localhost:3000/#/oauth/login
In v1.8.5, /oauth/auth endpoint redirects to "http://localhost:3000/?login_challenge=48ef86fa7f8142099aacbd172628f708#/oauth/login"
However, in v1.9.2 /oauth/auth endpoint redirects to "http://localhost:3000/%23/oauth/login?login_challenge=48ef86fa7f8142099aacbd172628f708"

**Reproducing the bug**

1. start local hydra server
```
#!/bin/sh
 
export URLS_LOGIN=http://localhost:3000/#/oauth/login
export URLS_CONSENT=http://localhost:3000/#/oauth/consent
export DSN=memory
export OIDC_SUBJECT_IDENTIFIERS_SUPPORTED_TYPES=public
export SERVE_COOKIES_SAME_SITE_MODE=Lax
export OAUTH2_EXPOSE_INTERNAL_ERRORS=true
export LOG_LEAK_SENSITIVE_VALUES=true
export LOG_LEVEL=debug
 
echo "Starting Hydra ..."
hydra serve all --dangerous-force-http
```
2. Create a client with
```
curl -L -X POST 'http://localhost:4445/clients' \
-H 'Accept: application/json' \
-H 'Content-Type: application/json' \
--data-raw '{
    "client_id": "demo",
    "client_secret": "password",
    "redirect_uris": [
        "http://localhost:8080/callback",
        "https://oauth.pstmn.io/v1/callback"
    ],
    "grant_types": [
        "authorization_code"
    ]
 }'
```
3. Initiate authorization code flow with
```
curl -L -X GET 'http://localhost:4444/oauth2/auth?response_type=code&state=123456&client_id=demo&redirect_uri=https%3A%2F%2Foauth.pstmn.io%2Fv1%2Fcallback'
```
The response header "Location" value is `http://localhost:3000/%23/oauth/login?login_challenge=48ef86fa7f8142099aacbd172628f708`

_Server logs_

No error

_Server configuration_

None

**Expected behavior**

Expect the redirect behavior remains compatible to v1.8.5 `http://localhost:3000/?login_challenge=48ef86fa7f8142099aacbd172628f708#/oauth/login`. So the angular login UI can remain functional after upgrade post v1.8.5.

**Environment**

- Version:    v1.9.2
Git Hash:   f0580e2581e202ec7299f45822db37228228aee9
Build Time: 2021-01-29T15:02:55Z
- Environment: Debian, Windows

### PR DESCRIPTION
**Pull request by simpleway**
allow #fragment in configured url to keep backwards compatibility. Close #2363

## Related issue

@aeneasr, this for keep backwards compatibility prior to v1.9.x.

## Proposed changes

Since the login, consent urls are configured by server owner, redirect to login or consent ui built with SPA should be allowed, although not recommended.

## Checklist

- [ x ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ x ] I have read the [security policy](../security/policy).
- [ x ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Comment by aeneasr** [comment]
Could you please add a test case for this? :)

**Comment by aeneasr** [comment]
Are you still up for the changes? :) If you need any help, let us know!

**Comment by simpleway** [comment]
added tests to the branch.

**Review by aeneasr — APPROVED** [review]
Awesome, thank you! 🎉 Your contribution makes Ory better :)

File: driver/config/provider_test.go [hunk 1]
```diff
@@ -314,3 +314,43 @@ func TestViperProviderValidates(t *testing.T) {
...
 }
+
+func TestSetPerm(t *testing.T) {
+	f, e := ioutil.TempFile("", "test")
+	require.NoError(t, e)
+	path := f.Name()
+
+	// We cannot test setting owner and group, because we don't know what the
+	// tester has access to.
+	_ = (&UnixPermission{
```
Comment (aeneasr) [inline driver/config/provider_test.go:325]:
```suggestion
	_ = (&configx.UnixPermission{
```

**Pull request merged**